Firestore/core/src/firebase/firestore/api/firestore.mm

```diff
@@ -61,31 +61,35 @@
       persistence_key_{std::move(persistence_key)},
       worker_queue_{std::move(worker_queue)},
       extension_{extension} {
-  settings_ = [[FIRFirestoreSettings alloc] init];
 }
 
 AsyncQueue* Firestore::worker_queue() {
   return [client_ workerQueue];
 }
 
-FIRFirestoreSettings* Firestore::settings() const {
+const Settings& Firestore::settings() const {
   std::lock_guard<std::mutex> lock{mutex_};
-  // Disallow mutation of our internal settings
-  return [settings_ copy];
+  return settings_;
 }
 
-void Firestore::set_settings(FIRFirestoreSettings* settings) {
+void Firestore::set_settings(Settings&& settings) {
```

Comment:
In general making parameters like these rvalue-references is unfriendly to callers. For example, a test might configure these settings once in a shared constant, and this signature would require that test to explicitly copy to make this call.

rvalue-references parameters are useful when you're explicitly marking a method as a sink that will consume or take ownership of the argument. They should otherwise be avoided.

In this case it's not critical that we avoid copying the Settings so passing by value gives the caller the flexibility to copy if it wants but move if it can.